src/Microsoft.Azure.WebJobs.Host/Bindings/BindingProviders/BindToStreamBindingProvider.cs

```diff
@@ -0,0 +1,492 @@
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        public Task<IBinding> TryCreateAsync(BindingProviderContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            var parameter = context.Parameter;
+            var typeUser = parameter.ParameterType;
+
+            var attributeSource = TypeUtility.GetResolvedAttribute<TAttribute>(parameter);
+
+            // Stream is either way; all other types are known. 
+            FileAccess? actualAccess = GetFileAccessFromAttribute(attributeSource);
+
+            Type argHelperType;
+            bool isRead;
+            if (typeUser == typeof(Stream))
+            {
+                if (!actualAccess.HasValue)
+                {
+                    throw new InvalidOperationException("When binding to Stream, the attribute must specify a FileAccess direction.");
+                }
+                switch (actualAccess.Value)
+                {
+                    case FileAccess.Read:
+                        isRead = true;
+                        
+                        break;
+                    case FileAccess.Write: 
+                        isRead = false;
+                        break;
+
+                    default:
+                        throw new NotImplementedException("ReadWrite access is not supported. Pick either Read or Write.");
+                }
+                argHelperType = typeof(StreamValueProvider);
+            }
+            else if (typeUser == typeof(TextReader))
+            {
+                argHelperType = typeof(TextReaderValueProvider);
+                isRead = true;
+            }
+            else if (typeUser == typeof(String))
+            {
+                argHelperType = typeof(StringValueProvider);
+                isRead = true;
+            }
+            else if (typeUser == typeof(byte[]))
+            {
+                argHelperType = typeof(ByteArrayValueProvider);
+                isRead = true;
+            }
+            else if (typeUser == typeof(TextWriter))
+            {
+                argHelperType = typeof(TextWriterValueProvider);
+                isRead = false;
+            }
+            else if (typeUser == typeof(String).MakeByRefType())
+            {
+                argHelperType = typeof(OutStringValueProvider);
+                isRead = false;
+            }
+            else if (typeUser == typeof(byte[]).MakeByRefType())
+            {
+                argHelperType = typeof(OutByteArrayValueProvider);
+                isRead = false;
+            }
+            else
+            {
+                // Totally unrecognized. Let another binding try it. 
+                return Task.FromResult<IBinding>(null);
+            }
+
+            VerifyOrThrow(actualAccess, isRead);
+            if (!IsSupportedByRule(isRead))
```

Comment:
Yes. Because if the rule only supports Read, then it's possible there's another rule that supports Write. IE, we can't assert it's an error. Whereas a conflict between #2 + #3 is definitely an error. 

---
In reply to: [145215473](https://github.com/Azure/azure-webjobs-sdk/pull/1387#discussion_r145215473) [](ancestors = 145215473)